### PR TITLE
Document @required

### LINKED ISSUE #886: Document @required from the meta library
I couldn't find this feature documented anywhere on www. Not in the library tour and not in the "commonly used library" page. It's used heavily in Flutter. It's even mentioned in [What's your Favorite Dart Feature?](https://news.dartlang.org/2016/10/whats-your-favorite-dart-feature.html)

### PR DESCRIPTION
**Pull request by chalin**
Fixes #886 

Staged, see:

- The last paragraph of https://sz-www-1.firebaseapp.com/guides/language/language-tour#functions
- https://sz-www-1.firebaseapp.com/guides/language/language-tour#optional-named-parameters

cc @Sfshaza @filiph

### CODE REVIEW

**Comment by Sfshaza** [comment]
LGTM!  I would merge, but I'll wait for Kathy to review.

File: examples/misc/lib/language_tour/functions.dart [hunk 1]
```diff
@@ -1,5 +1,7 @@
 // ignore_for_file: unused_element, type_annotate_public_apis
 
+import 'package:meta/meta.dart';
```
Comment (Sfshaza) [inline examples/misc/lib/language_tour/functions.dart]: I wonder if they'll ever move required to the core and out of meta.

File: src/_guides/language/language-tour.md [hunk 2]
```diff
@@ -911,11 +912,24 @@ to specify named parameters:
...
+
+You can annotate a named parameter in any Dart code (not just Flutter) with
+[@required][] to indicate that it is a _required_ parameter. For example:
+
+<?code-excerpt "misc/lib/language_tour/functions.dart (required-named-parameters)" replace="/@required/[!$&!]/g"?>
+{% prettify dart %}
+const Scrollbar({Key key, [!@required!] Widget child})
 {% endprettify %}
 
+When a `Scrollbar` is constructed, the analyzer will report an issue if the
```
Comment (Sfshaza) [inline src/_guides/language/language-tour.md]: will report => reports

https://developers.google.com/style/tense

Comment (chalin) [reply src/_guides/language/language-tour.md]: Done, thanks.

File: examples/misc/lib/language_tour/functions.dart (hunk 1 above)
Comment (chalin) [reply examples/misc/lib/language_tour/functions.dart]: Ah, right. I've added a sentence about imports. PTAL

**Comment by chalin** [comment]
Updates done, staging server refreshed.

File: src/_guides/language/language-tour.md [hunk 3]
```diff
@@ -911,11 +912,28 @@ to specify named parameters:
...
+<?code-excerpt "misc/lib/language_tour/functions.dart (required-named-parameters)" replace="/@required/[!$&!]/g"?>
+{% prettify dart %}
+const Scrollbar({Key key, [!@required!] Widget child})
+{% endprettify %}
+
+When a `Scrollbar` is constructed, the analyzer reports an issue when the
+`child` argument is absent.
+
+[Required][@required] is defined in the [meta][] package. Import
+`package:meta/meta.dart` directly, or other package that exports `meta` such
```
Comment (kwalrath) [inline src/_guides/language/language-tour.md]: Import -> Either import
other -> import another
 such -> , such

File: examples/misc/lib/language_tour/functions.dart (hunk 1 above)
Comment (chalin) [reply examples/misc/lib/language_tour/functions.dart]: Kathy gave a thumbs up, so I'll merge. We can always revisit this as needed.

**Pull request merged**